azure/services/groups/spec.go

```diff
@@ -18,16 +18,17 @@ package groups
 
 import (
 	"context"
-	"fmt"
 
 	asoresourcesv1 "github.com/Azure/azure-service-operator/v2/api/resources/v1api20200601"
-	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/utils/ptr"
 	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
 	"sigs.k8s.io/cluster-api-provider-azure/azure/services/aso"
 )
 
+// ASOType aliases the ASO object type managed by the service.
+type ASOType = *asoresourcesv1.ResourceGroup
```

Comment:
nice

Reply:
although it looks like we're still using `asoresourcesv1.ResourceGroup` directly in some places, isn't it potentially more confusing if it's a mix of both?

Reply:
I think the only place where it's necessary to use the regular name is in the `GroupScope` interface definition in groups.go. Using `ASOType` there results in an import cycle in the generated mocks.

I think I might slightly prefer the full name anyway since I think that's a bit more clear. I expect the compiler would catch any copy-paste mistakes.

Reply:
I meant usages in this file specifically https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3997/files#diff-ba2c0ee6f840a091bd5c2362b53be5d53f384bcb347c7f11feff1c9c0ba9351bL57

Reply:
Do you prefer using the `ASOType` alias everywhere or the full `*asoresourcesv1.ResourceGroup` everywhere?

Reply:
I don't have a strong preference, my only preference is consistency